src/dnne-gen/Generator.cs

```diff
@@ -1025,11 +1030,20 @@ public string GetPinnedType(string elementType)
 
             public string GetPointerType(string elementType)
             {
+                mLastUnsupportedPrimitiveType = null;
                 return elementType + "*";
             }
 
             public string GetPrimitiveType(PrimitiveTypeCode typeCode)
             {
+                ThrowIfUnsupportedLastPrimitiveType();
+
+                if (typeCode == PrimitiveTypeCode.Char)
+                {
+                    mLastUnsupportedPrimitiveType = typeCode;
+                    return "wchar_t";
```

Comment:
Like so?

```C
#ifdef DNNE_WINDOWS
    ...
    #define DNNE_WCHAR wchar_t
#else
    ...
    #define DNNE_WCHAR uint16_t
#endif
```

```CSharp
if (typeCode == PrimitiveTypeCode.Char)
{
    // Record the current type here with the expectation
    // it will be of pointer type to Char, which is supported.
    this.lastUnsupportedPrimitiveType = typeCode;
    return "DNNE_WCHAR";
}
```

Reply:
Yes.